internal/provider/vdc/acl_resource.go

```diff
@@ -0,0 +1,376 @@
+	}
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *aclResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state *aclResourceModel
+
+	// Get current state
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Init resource
+	resp.Diagnostics.Append(r.Init(ctx, state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if _, err := r.vdc.DeleteControlAccess(true); err != nil {
+		resp.Diagnostics.AddError("Error deleting control access", err.Error())
+		return
+	}
+}
+
+//go:generate go run github.com/FrangipaneTeam/tf-doc-extractor@latest -filename $GOFILE -example-dir ../../../examples -resource
+func (r *aclResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("vdc"), req, resp)
+}
+
+func sharedSetToAccessControl(client *govcd.VCDClient, org *govcd.AdminOrg, input []acl.SharedWithModel) ([]*govcdtypes.AccessSetting, []*acl.SharedWithModel, error) {
+	var output []*govcdtypes.AccessSetting
+	var outputModel []*acl.SharedWithModel
+
+	for _, item := range input {
+		var subjectHref string
+		var subjectType string
+		var subjectName string
+		var oModel *acl.SharedWithModel
+
+		if !item.UserID.IsNull() && !item.UserID.IsUnknown() {
+			userID := item.UserID.ValueString()
+			user, err := org.GetUserById(userID, false)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error retrieving user %s: %s", userID, err)
+			}
+			subjectHref = user.User.Href
+			subjectType = user.User.Type
+			subjectName = user.User.Name
+
+			oModel = &acl.SharedWithModel{
+				UserID:      types.StringValue("urn:vcloud:user:" + common.ExtractUUID(subjectHref)),
+				SubjectName: types.StringValue(subjectName),
+			}
+		} else if !item.GroupID.IsNull() && !item.GroupID.IsUnknown() {
+			groupID := item.GroupID.ValueString()
+			group, err := org.GetGroupById(groupID, false)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error retrieving group %s: %s", groupID, err)
+			}
+			subjectHref = group.Group.Href
+			subjectType = group.Group.Type
+			subjectName = group.Group.Name
+			oModel = &acl.SharedWithModel{
+				GroupID:     types.StringValue("urn:vcloud:group:" + common.ExtractUUID(subjectHref)),
+				SubjectName: types.StringValue(subjectName),
+			}
+		}
+
+		accessLevel := item.AccessLevel.ValueString()
+
+		output = append(output, &govcdtypes.AccessSetting{
+			Subject: &govcdtypes.LocalSubject{
+				HREF: subjectHref,
+				Name: subjectName,
+				Type: subjectType,
+			},
+			ExternalSubject: nil,
+			AccessLevel:     accessLevel,
+		})
+		oModel.AccessLevel = types.StringValue(accessLevel)
+		outputModel = append(outputModel, oModel)
+	}
+	return output, outputModel, nil
+}
+
+func accessControlListToSharedSet(input []*govcdtypes.AccessSetting) ([]acl.SharedWithModel, error) {
```

Comment:
Why not return `diag.Diagnostics` rather than an `error`

Reply:
K.I.S.S ?